### PR TITLE
Back merge Release/4.8.x november 22 to master

### PR DESCRIPTION
**Pull request by haraldloesing**

### CODE REVIEW

**Review by ndinat — APPROVED** [review]

File: src/main/java/ch/admin/bag/covidcertificate/domain/AuthHolderRepository.java [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
 import java.util.UUID;
 
 @Repository
 public interface AuthHolderRepository extends JpaRepository<AuthHolder, UUID> {
+    @Query(value = "select ah from AuthHolder ah " +
+            "where ah.code = :code ")
+    AuthHolder findByCode(@Param("code") String code);
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/domain/AuthHolderRepository.java:15]: jpa does not need a query here - findByCode gets translated natively to the right query (see https://www.bezkoder.com/jpa-repository-query/#JPA_find_by_field)

File: src/main/java/ch/admin/bag/covidcertificate/domain/DisplayNameModificationRepository.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.util.Optional;
+
+@Repository
+public interface DisplayNameModificationRepository extends JpaRepository<DisplayNameModification, String> {
+
+    @Query(value = "select dnm from DisplayNameModification dnm " +
+            "where dnm.code = :code " +
+            "and dnm.entityType = :entityType ")
+    Optional<DisplayNameModification> findByCodeAndEntityType(@Param("code") String code, @Param("entityType") EntityType entityType);
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/domain/DisplayNameModificationRepository.java:17]: jpa does not need a query here - findByCodeAndEntityType gets translated natively to the right query (see https://www.bezkoder.com/jpa-repository-query/#JPA_find_by_multiple_Columns)

File: src/main/java/ch/admin/bag/covidcertificate/domain/ProphylaxisRepository.java [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
 import java.util.UUID;
 
 @Repository
 public interface ProphylaxisRepository extends JpaRepository<Prophylaxis, UUID> {
+    @Query(value = "select p from Prophylaxis p " +
+            "where p.code = :code ")
+    Prophylaxis findByCode(@Param("code") String code);
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/domain/ProphylaxisRepository.java:15]: jpa does not need a query here - findByCode gets translated natively to the right query (see https://www.bezkoder.com/jpa-repository-query/#JPA_find_by_field)

File: src/main/java/ch/admin/bag/covidcertificate/domain/VaccineImportControlRepository.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import java.time.LocalDate;
+import java.util.Optional;
+
+@Repository
+public interface VaccineImportControlRepository extends JpaRepository<VaccineImportControl, String> {
+
+    @Query(value = "select vic from VaccineImportControl vic " +
+            "where vic.importDate <= :importDate " +
+            "and vic.done = false")
+    Optional<VaccineImportControl> findByImportDate(@Param("importDate") LocalDate importDate);
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/domain/VaccineImportControlRepository.java:17]: jpa does not need a query here - findByImportDateLessThanEqualAndDoneFalse gets translated natively to the right query (see https://www.bezkoder.com/jpa-repository-query/#JPA_Repository_Query_with_Comparison and https://www.bezkoder.com/jpa-repository-query/#JPA_Boolean_Query)

File: src/main/java/ch/admin/bag/covidcertificate/domain/VaccineRepository.java [hunk 5]
```diff
@@ -36,4 +37,10 @@ public interface VaccineRepository extends JpaRepository<Vaccine, UUID> {
             "order by v.display asc")
     List<Vaccine> findAllPlatformApiActive();
 
+    @Query(value = "select v from Vaccine v " +
+            "where v.code = :code ")
+    List<Vaccine> findByCode(@Param("code") String code);
+
+    @Query(value = "select distinct v.code from Vaccine v ")
+    List<String> findAllCodes();
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/domain/VaccineRepository.java:45]: See other comments on this subject.

File: src/main/java/ch/admin/bag/covidcertificate/domain/ValueSetUpdateLogRepository.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+import java.util.List;
+import java.util.UUID;
+
+@Repository
+public interface ValueSetUpdateLogRepository extends JpaRepository<ValueSetUpdateLog, UUID> {
+
+    @Query(value = "select vsul from ValueSetUpdateLog vsul " +
+            "where vsul.code = :code ")
+    ValueSetUpdateLog findByCode(@Param("code") String code);
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/domain/ValueSetUpdateLogRepository.java:16]: See other comments on this subject.

**Review by Armin-Isenring-Bit — COMMENTED** [review]
Check the JPA's native method interpretation and judge on your own about the simplest solution.

**Review by YanikFenner — COMMENTED** [review]
I.O.

**Review by YanikFenner — APPROVED** [review]

**Pull request merged**